devel/tortoisehg/Portfile

```diff
@@ -18,23 +18,30 @@ description         A set of graphical tools for Mercurial
 long_description    A set of graphical tools for the Mercurial distributed \
                     source control management system.
 
-checksums           rmd160  9df80404cefa5567264913ea93ad778a4d971c70 \
-                    sha256  7c6d58b34678f237b0683dcac38707c91e10b46a28f918888fe6bb2c005c762c \
-                    size    8211370
+checksums           rmd160  bb915e3c207f82e19749b680bb0740c3fa9e5ca9 \
+                    sha256  5ae2cf9065bfbbd1491ff8bd4e895da92996da8622ef356a13b734e65199e62e \
+                    size    8219445
 
 python.default_version 27
 
-# Can use either mercurial or mercurial-devel.
-depends_lib         path:bin/hg:mercurial \
-                    port:py${python.version}-pyqt4 \
-                    port:py${python.version}-qscintilla-qt4 \
+# we use qt5 only if macOS is 10.12 or later
+# as defined on qt5 port
+if { ${os.major} < 16 } {
+    set qt_version 4
+} else {
+    set qt_version 5
+}
+
+depends_lib         port:mercurial \
+                    port:py${python.version}-pyqt${qt_version} \
+                    port:py${python.version}-qscintilla-qt${qt_version} \
                     port:py${python.version}-iniparse
 
 depends_build       port:py${python.version}-sphinx
 
 post-extract {
     copy ${filespath}/config.py ${worksrcpath}/tortoisehg/util/
-    reinplace -W ${worksrcpath} "s,pyrcc4,pyrcc4-2.7," setup.py
+    reinplace -W ${worksrcpath} "s,pyrcc${qt_version},pyrcc${qt_version}-2.7," setup.py
```

Comment:
Actually it should not matter here if we patch both 4 and 5, but this just got me thinking that we might want to do it in a patchfile?

Reply:
In this case I am not sure. True that probably we don't catch the error, maybe the build fail I don't know, but it's only one occurrence and it is more resistant to changes of setup.py